nuxt.config.ts

```diff
@@ -33,6 +33,18 @@ export default defineNuxtConfig({
       htmlAttrs: {
         lang: 'en',
       },
+      script: [
+        {
+          type: 'text/javascript',
+          innerHTML: `
+          if (localStorage.theme === "dark" || (!('theme' in localStorage) && window.matchMedia("(prefers-color-scheme: dark)").matches)) {
+            document.documentElement.classList.add('dark')
+          } else {
+            document.documentElement.classList.remove('dark')
+          }
+        `,
+        },
+      ],
```

Comment:
Instead of having `useThemeHead` and using it in `default` layout and `error`, I'm just setting it here. This also improves the situation with the flicker before the theme kicks in.